language/stdlib/modules/event.mvir

```diff
@@ -0,0 +1,92 @@
+// This module is used for emitting events into the event_store.
+module Event {
+    import 0x0.AddressUtil;
+    import 0x0.BytearrayUtil;
+    import 0x0.Hash;
+    import 0x0.U64Util;
+
+    // A resource representing the counter used to generate uniqueness under each account. There won't be destructor for
+    // this resource to guarantee the uniqueness of the generated handle.
+    resource HandleIdGenerator {
+        // A monotonically increasing counter
+        counter: u64,
+    }
+
+    // A handle for a event
+    resource Handle {
+        // HandleIdGenerator for this event stream.
+        counter: u64
+        // A globally unique ID for this event stream.
+        guid: bytearray,
+    }
+
+    // Get the reference to HandleIdGenerator stored under sender's account.
+    get_counter(): &mut R#Self.HandleIdGenerator {
```

Comment:
Sorry I missed this on my last review. 

`get_counter` can't exist as you have written it in the future. We are banning the ability to return global references, and we shouldn't add any new call sites